### PR TITLE
upgrade etcd dependencies to v3.5.6

### PR DESCRIPTION
**Pull request by pacoxu**
#### What type of PR is this?
/kind cleanup

#### What this PR does / why we need it:
follow up of 
- https://github.com/etcd-io/etcd/pull/14489
- https://github.com/kubernetes/kubernetes/pull/114912

upgrade etcd server dependency to v3.5.6 to remove  `github.com/form3tech-oss/jwt-go`.
#### Which issue(s) this PR fixes:

Fixes None

#### Special notes for your reviewer:
/cc @dims @liggitt

#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

File: vendor/go.etcd.io/etcd/client/v3/watch.go [hunk 1]
```diff
@@ -580,6 +589,26 @@ func (w *watchGrpcStream) run() {
 
 			switch {
 			case pbresp.Created:
+				cancelReasonError := v3rpc.Error(errors.New(pbresp.CancelReason))
```
Comment (liggitt) [inline vendor/go.etcd.io/etcd/client/v3/watch.go:592]: still waiting on a revert of in the 3.5 stream before pulling in an etcd update

xref https://github.com/kubernetes/kubernetes/pull/114403#discussion_r1045906333

Comment (pacoxu) [reply vendor/go.etcd.io/etcd/client/v3/watch.go:592]: /hold

Comment (dims) [reply vendor/go.etcd.io/etcd/client/v3/watch.go:592]: also https://github.com/kubernetes/kubernetes/pull/114403 :)

**Pull request closed without merging**